src/main/java/io/temporal/samples/retryonsignalinterceptor/MyActivityImpl.java

```diff
@@ -0,0 +1,41 @@
+/*
+ *  Copyright (c) 2020 Temporal Technologies, Inc. All Rights Reserved
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.samples.retryonsignalinterceptor;
+
+import io.temporal.failure.ApplicationFailure;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class MyActivityImpl implements MyActivity {
+
+  private final AtomicInteger count = new AtomicInteger();
```

Comment:
I would look at the activity info attempt instead.
The way it is currently implemented will of course not work in a "real" activity since it may be executed on different workers and I'd rather not encourage this practice.